### PR TITLE
Prepare minor version upgrade and Fix text cannot convert to datetime in DATEADD()

### PR DESCRIPTION
**Pull request by ZhenyeLee**
Create upgrade script for 2.2 to 2.3
Bump up TSQL extension minor version number
Rename latest folder to 14_5
Create a new latest folder, add schedule file
Add 14.5->lastest upgrade path in upgrade-test-configuration.yml

For BABEL-1493:
DATEADD() function raises an error when input is text or string. In this PR, input string is first checked to see if it is in date format using isdate() function. If input text is a date, then it will convert to datetime datatype and using exist function to get output.

Task: No JIRA for prepare minor version upgrade and BABEL-1493
Signed-off-by: Zhenye Li <zhenye@amazon.com>

### Description

[Describe what this change achieves]
 
### Issues Resolved

[List any issues this PR will resolve]

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-prepare.mix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+-- tsql
```
Comment (rishabhtanwar29) [inline test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-prepare.mix]: This doesn't seem to be right, why these prepare verify files are not inside `preparation` directory. There should a preparation directory for 14_5 and all the prepare files should go there.

File: test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-prepare.out [hunk 2]
```diff
@@ -0,0 +1,21 @@
+-- tsql
```
Comment (rishabhtanwar29) [inline test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-prepare.out]: Again, all the expected files should be inside `expected` directory, not here.

File: test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-verify.mix [hunk 3]
```diff
@@ -0,0 +1,9 @@
+-- tsql
```
Comment (rishabhtanwar29) [inline test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-verify.mix]: These verification and cleanup files should be inside `verification_cleanup` directory.

File: .github/template/version-branch-template.yml [hunk 4]
```diff
@@ -13,6 +13,9 @@
 '14.3':
   engine_branch: BABEL_2_1_STABLE__PG_14_3
   extension_branch: BABEL_2_1_STABLE
+'14.5':
+  engine_branch: latest
+  extension_branch: latest
```
Comment (rishabhtanwar29) [inline .github/template/version-branch-template.yml:18]: `latest` is okay for now but we need to change these branches to stable once they are cut.

File: test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-verify.mix (hunk 3 above)
Comment (ZhenyeLee) [reply test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-verify.mix]: Those files are not in neither the verification or cleanup folder right now in GitHub. Do you think we need to move them to verification_cleanup folder?

Comment (rishabhtanwar29) [reply test/JDBC/upgrade/14_5/SYSTEM_USER_dep-vu-verify.mix]: Oh ok, looks like these files have been added to wrong place from the beginning itself, but the issue is that these tests won't get executed if we keep them here. I think you can keep these here and ask author to move those files in correct folder.

**Pull request closed without merging**